torch/onnx/symbolic_opset9.py

```diff
@@ -362,12 +362,18 @@ def _maybe_cast_reduce_op_input(g, self):
     return self
 
 
+def _handle_reduce_dim_none(g, self, op_name):
+    if len(list(self.node().inputs())) > 0:
```

Comment:
What is this condition checking here?

Can we move this into helper?

Reply:
TestOperators.test_sum/TestOperators.test_mean/TestOperators.test_prod tests will also go through this symbolic function. For these 3 tests, we need to keep the attribute keepdims to 0 as before, while we need to set keepdims to 1 for the scenario mentioned in this PR.

Based on above, this checking is to distinguish the scenario in this PR from the scenario of those 3 TestOperators tests.

This _handle_reduce_dim_none() is only for one situation in _reduce_op_symbolic() method, and it could not be used commonly. So I think it'd better to be closed to where _reduce_op_symbolic() lives in. Not necessary to be in helper.

Reply:
synced offline with Jay, this condition will be updated to check for empty tensor input (shape contains only 0).